### DISCUSSION TITLE
[Feature Request] Mark dates in the calendar component

### DISCUSSION THREAD

**Discussion by kjorand** [heroui-inc/heroui#4727] [Feature requests]
### Is your feature request related to a problem? Please describe.

Wanted to use the calendar to show availability. BUT ... still be able to select any date to see what's going on on that day.

### Describe the solution you'd like

An additionnal prop something like "isSpecialDate" => then we could add some custom formating to the date in the calendar, either strikethrough like unavailable ... or maybe with a faded background or somthing (already more generic feature than strikethroug) ... 
OR even better and most generic : a "dateFormater" prop to which we could pass a callback (just like isUnavailable) but which callback should return some specific formatting like
```ts
{
    strikethrough?: boolean,
    color?: DateProps["color"], // or similar
    faded?: boolean, // or whatever other appearance is easy and makes sense to offer
    // ... other ???
}
```
... or actually just returning an array or string of classNames would be great already

Would be most generic and even allow the callback to differentiate occupency let's say : 
- "normal" => available whole day => {}
- "completely unavailable" => {strikethrough: true, faded: true}
- "partially unavailable" => {faded: true}
- "special" => {color: "warning"}
- .... 

### Describe alternatives you've considered

Already described as first part above ... Still believe the last and most generic solution would be best ! (my use-case actually involves partial unavailability and complete unavailability)

### Screenshots or Videos

![image](https://github.com/user-attachments/assets/4483ace6-b230-47c6-94ea-8203cdda05cb)

Here I just edited the example from the docs manually with the developper tools of the browser, adding text-warning class ... works pretty well ... 

And huh ... I'll gladly answer questions if anything is unclear ... but right now I don't see anything else ...

**Comment by linear** [comment]
<p><a href="https://linear.app/nextui-inc/issue/ENG-1431/[feature-request]-mark-dates-in-the-calendar-component">ENG-1431 [Feature Request] Mark dates in the calendar component</a></p>

**Comment by joshclement0** [comment]
I would also like to do something like this:
![image](https://github.com/user-attachments/assets/bf3693e3-13cf-474f-822e-4625cc577e2c)
With a custom component behind the number

it seems these two problems might be separate but solved in the same manner. i.e. a callback that is passed down from the user to  add a class name or component based on the specific day (d?:Date) => style or (d:Date) => html
